packages/hasher/src/yarnLock.ts

```diff
@@ -0,0 +1,42 @@
+import * as findUp from "find-up";
+import * as fs from "fs-extra";
+import * as lockfile from "@yarnpkg/lockfile";
+
+import { Dependencies } from "./resolveExternalDependencies";
+import { nameAtVersion } from "./helpers";
+
+type YarnLockDependency = {
+  version: string;
+  dependencies?: Dependencies;
+};
+
+export type ParsedYarnLock = {
+  type: "success" | "merge" | "conflict";
+  object: {
+    [key in string]: YarnLockDependency;
+  };
+};
+
+export async function parseLockFile(
```

Comment:
I'm exporting this function to only have to parse the lock file once. Otherwise, I would need to use some kind of singleton or module-scope variable to keep track of the parsed lock file. I would rather not do that. Also, it makes it easier to test.